sdk/src/manifest.rs

```diff
@@ -644,7 +644,8 @@ impl Manifest {
         // convert the manifest to a store
         let mut store = self.to_store()?;
         // sign and write our store to to the output image file
-        store.save_to_asset(source_path.as_ref(), signer, dest_path.as_ref())?;
+        // set source and dest to same path to avoid an unnecessary copy since we already copied above
+        store.save_to_asset(dest_path.as_ref(), signer, dest_path.as_ref())?;
```

Comment:
There's problem here. We only copy the source to the dest above if the dest does not already exist. If it does, then the two are different files. So if we want this optimization we have to check for the case where dest is a different file.

Reply:
FYI, I have a plan to redo the command line options on c2patool so that this kind of thing will be clear.

Reply:
maybe change the value for source_path only if you copy the source to the dest?

Reply:
If the only reason you are doing the copy is to get the info.  Why no just set the info from the source when the dest does not exit.  Then you have no copy code here:

 if !dest_path.as_ref().exists() {
         // first add the information about the target file
        self.set_asset_from_path(source_path.as_ref());
 } else {
        // first add the information about the target file
        self.set_asset_from_path(dest_path.as_ref());
}

Reply:
I changed  to logic to  make this more clear